README.md

```diff
@@ -1,2 +1,80 @@
-# dishes
+# dishes #
+
 Chef cookbook to clean up recipes after removal
+
+## Motivation ##
+
+Chef does not generally support an "undo" feature for recipes.  Once one executes, there isn't an easy way to return a node to it's pre-recipe state.  To solve the problem systematically looks rather hard, so Chef is perhaps wise not to try.  There is however at least one scenario where a feasible, limited "undo" would be very helpful.
+
+A recurring problem that can afflict distributed services is the [_split brain_ scenario][split-brain-wikipedia].  Though it can emerge in many different contexts, a common cause is that some daemon is running unexpectedly when we would really prefer it to be off.
+
+The "right" way to avoid split brains caused this way is of course to ensure that the offending daemon is turned off to start with.  However if the daemon is managed with Chef then this can be difficult.  Particularly in a bare-metal cluster with long-lived nodes, a node's run list may evolve over time and the recipe controlling the daemon might be removed.  (A cluster running in a cloud environment would be better served by destroying and recreating the nodes, avoiding the evolving run lists completely.)  An unwary administrator might see later that the node does not have the service in its run list and assume the daemon is not installed (let alone running.)
+
+Chef itself does not offer a way to cleanup removed entries from a run list as a first class feature, perhaps because we can emulate this ourselves.  For example, one can imagine our administrator adding a `disable_service_daemon` "cleanup" recipe to the node at the same time that the original daemon recipe is removed.
+
+That works, but it requires that the clean up recipe exists and that the administrator is aware of the issue and so can take action.  This however scales poorly: consider that the offending daemons might be brought as transitive cookbook dependencies.  It would better if Chef could do this for us.
+
+It turns out that it can, with some help.  This cookbook supplies two LWRPs named `dishes_dish` and `dishes_sink`, which work together to enable Chef (and recipe authors) to react to recipes removed from a run list by including corrosponding "cleanup" recipes.
+
+[split-brain-wikipedia]: https://en.wikipedia.org/wiki/Split-brain_(computing)
+
+## Usage ##
+
+```ruby
+# Tell dishes_sink to include `disable_daemon` if `daemon` is not in the run list
+dishes_dish 'cookbook::daemon' do
+  action :use
+  wash_recipe 'cookbook::disable_daemon'
+  sink 'default'
+end
+
+# Include the wash recipes of all dishes whose recipe is not in the run list
+dishes_sink 'default'
+```
+
```

Comment:
Might want to explain what happens in this scenario. I'm assuming from the previous text that the wash_recipe will get to work when cookbook::daemon is removed from the run_list. We might want to be explicit about this here though.